src/tools/bin/plot_chargeConservation_overTime.py

```diff
@@ -0,0 +1,209 @@
+# get all hdf5 output files
+def get_list_of_hdf5_files(base_directory):
+    """
+    Returns a list of hdf5 files (`h5_*.h5`)
+    listed in sub-directory `simOutput`
+
+    Parameters:
+    base_directory: string
+        directory path where to find simOutput
+
+    Return:
+    list of strings with hdf5 file names found
+    """
+    h5_list = [] # empty list for hdf5 files
+    h5_dir = base_directory + "/simOutput/"
+    if not os.path.isdir(h5_dir):
+        raise Exception(("Error: {} does not contain"+
+                         " a simOutput directory").format(directory))
+
+    for filename in os.listdir(h5_dir):
+        if os.path.isfile(h5_dir+filename):
+            if re.search("h5_.*.h5", filename):
+                h5_list.append(h5_dir + filename)
+    return h5_list
+
+
+def deviation_charge_conservation(h5file):
+    """
+    read field data from hdf5 files
+    compute d = div(E) - rho/epsilon_o0
+
+    Parameters:
+    h5file: file name
+        file name and path to hdf5 data from PIConGPU
+
+    Return:
+    list of floats: [timestep, max(abs(d)),
+        mean(abs(d)), std(d), norm]
+    """
+    # load hdf5 file
+    f = h5py.File(h5file, "r")
+
+    # read time step (python 2 and 3 save)
+    timestep = -1
+    for i in f['/data'].keys():
+        timestep = i
+
+    # load physics constants and simulation parameters
+    EPS0 = f["/data/{}".format(timestep)].attrs["eps0"]
+    CELL_WIDTH = f["/data/{}".format(timestep)].attrs["cell_width"]
+    CELL_HEIGHT = f["/data/{}".format(timestep)].attrs["cell_height"]
+    CELL_DEPTH = f["/data/{}".format(timestep)].attrs["cell_depth"]
+    CELL_VOLUME = CELL_WIDTH * CELL_HEIGHT * CELL_DEPTH
+
+    # load electric field
+    Ex = np.array(f["/data/{}/fields/FieldE/x".format(timestep)])
+    Ey = np.array(f["/data/{}/fields/FieldE/y".format(timestep)])
+    Ez = np.array(f["/data/{}/fields/FieldE/z".format(timestep)])
+
+    # load and add charge density
+    charge = np.zeros_like(Ex)
+    norm = 0.0
+    for field_name in f["/data/{}/fields/".format(timestep)].keys():
+        if field_name[0:8] == "Density_":
+            # load species density
+            species_Density = np.array(f["/data/{}/fields/".format(timestep) + field_name])
+            # choose norm to be the maximal charge density of all species
+            norm = np.max([norm, np.amax(np.abs(species_Density))])
+            # add charge density to total charge density
+            charge += species_Density
+
+    # close hdf5 file
+    f.close()
+
+    # compute divergence of electric field according to Yee scheme
+    div = ((Ex[1:, 1:, 1:] - Ex[1:, 1:, :-1])/CELL_WIDTH +
+           (Ey[1:, 1:, 1:] - Ey[1:, :-1, 1:])/CELL_HEIGHT +
+           (Ez[1:, 1:, 1:] - Ez[:-1, 1:, 1:])/CELL_DEPTH)
+
+    # compute difference between electric field divergence and charge density
+    diff = (div  - charge[1:, 1:, 1:]/EPS0)
+
+    return float(timestep), np.amax(np.abs(diff)), np.mean(np.abs(diff)), np.std(diff), norm
```

Comment:
here, I preferred the standard deviation of the entire data set `np.std(diff)` over the standard deviation of the mean value `np.std(diff)/np.product(diff.shape)` since I think a statement over the entire data set is more interesting than over the mean value alone.